code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/SmallImageCard.kt

```diff
@@ -50,16 +50,20 @@ fun SmallImageCard(
     style: SmallImageUIStyle,
     observer: AepUIEventObserver?,
 ) {
+    var isLoading by remember { mutableStateOf(true) }
     var imageBitmap by remember { mutableStateOf<Bitmap?>(null) }
     val imageUrl = if (isSystemInDarkTheme() && ui.getTemplate().image?.darkUrl != null)
         ui.getTemplate().image?.darkUrl else ui.getTemplate().image?.url
 
     LaunchedEffect(ui.getTemplate().id) {
         observer?.onEvent(UIEvent.Display(ui))
-        if (imageUrl != null) {
+        if (imageUrl.isNullOrBlank()) {
+            isLoading = false
+        } else {
             imageBitmap = withContext(Dispatchers.IO) {
                 UIUtils.downloadImage(imageUrl)
             }
+            isLoading = false
```

Comment:
If the algorithm is  
- `always expect an image and display loading indicator`
- `If image is null dismiss loading indicator, else download image`
- `dismiss loading indicator once image is downloaded`

should `isLoading` be set to false inside the coroutine?

Reply:
Changed `UIUtils.downloadImage(url: String)` to use ServiceProvider network service. Since that is already guaranteed to run on a different thread, removed switching content to IO coroutine here

Reply:
Thank you for the great explanation! Changed it to use a callback. I'm working on adding tests, facing some weird issues. I'll update the PR in a bit